src/Marten/Services/JsonNetSerializer.cs

```diff
@@ -61,54 +62,50 @@ public void Customize(Action<JsonSerializer> configure)
             _withTypes.TypeNameHandling = TypeNameHandling.Objects;
         }
 
-        public void ToJson(object document, TextWriter writer)
+        public void ToJson(object document, Stream stream)
         {
-            using var jsonWriter = new JsonTextWriter(writer)
+            using var jsonWriter = new JsonTextWriter(new StreamWriter(stream))
             {
                 ArrayPool = _jsonArrayPool,
                 CloseOutput = false,
                 AutoCompleteOnClose = false
             };
 
-
             _serializer.Serialize(jsonWriter, document);
 
-            writer.Flush();
+            jsonWriter.Flush();
         }
 
         public string ToJson(object document)
         {
-            var writer = new StringWriter();
-            ToJson(document, writer);
+            using var stream = new MemoryStream();
```

Comment:
Why are we having to write to a stream like this?

Reply:
I reverted those changes and brought back old implementation after changing method signature with DBReader etc.